lib/src/signed_video_h26x_auth.c

```diff
@@ -744,30 +783,43 @@ maybe_validate_gop(signed_video_t *self, h26x_nalu_t *nalu)
 
   svi_rc status = SVI_UNKNOWN;
   SVI_TRY()
-    SVI_THROW(prepare_for_validation(self));
+    // |first_validation_check| keeps track of first validation when looping through pending gops
+    bool first_validation_check = true;
+    // Loop through possible pending gops and validate them
+    for (int i = 0; i < nalu_list->gop_idx; i++) {
+      memcpy(gop_state, &nalu_list->gop_state_pending[i], sizeof(gop_state_t));
+      memcpy(
+          gop_info_detected, &nalu_list->gop_info_detected_pending[i], sizeof(gop_info_detected_t));
 
-    if (!gop_state->signing_present) {
-      verify_hashes_without_sei(self);
-      latest->authenticity = SV_AUTH_RESULT_NOT_SIGNED;
-    } else {
-      validate_authenticity(self);
-    }
+      if (!first_validation_check) gop_state->is_first_validation = false;
 
-    // The flag |is_first_validation| is used to ignore the first validation if we start the
-    // validation in the middle of a stream. Now it is time to reset it.
-    gop_state->is_first_validation = false;
+      SVI_THROW(prepare_for_validation(self));
 
-    // Reset the gop_state_t and gop_info_detected_t.
-    gop_state_reset(gop_state, gop_info_detected);
-    // If we find a pending SEI and it is the latest NALU the state should be
-    // AUTH_STATE_WAIT_FOR_NEXT_NALU.
-    h26x_nalu_list_item_t *sei = h26x_nalu_list_get_next_sei_item(nalu_list);
-    if (sei && (sei == self->nalu_list->last_item)) {
-      gop_info_detected->has_gop_sei = true;
-      gop_state->auth_state = AUTH_STATE_WAIT_FOR_NEXT_NALU;
+      if (!gop_state->signing_present) {
+        verify_hashes_without_sei(self);
+        latest->authenticity = SV_AUTH_RESULT_NOT_SIGNED;
+      } else {
+        validate_authenticity(self);
+      }
+
+      // The flag |is_first_validation| is used to ignore the first validation if we start the
+      // validation in the middle of a stream. Now it is time to reset it.
+      gop_state->is_first_validation = false;
+      first_validation_check = false;
+
+      // Reset the gop_state_t and gop_info_detected_t.
+      gop_state_reset(gop_state, gop_info_detected);
+      // If we find a pending SEI and it is the latest NALU the state should be
+      // AUTH_STATE_WAIT_FOR_NEXT_NALU.
```

Comment:
This special operation is not handled when storing gop_states on line 744-745. Is this case tested?
Try removing the first GOP in test lost_all_nalus_between_two_seis instead